tutorials/basic-topics/sorting/insertion-sort.md

```diff
@@ -11,4 +11,202 @@ keywords:
+
+We have figured out the solution of the problem:
+* Sort the given array using insertion sort (Refer to the algorithm section).
+* Take out the last two elements because those are the largest two elements in our array.
+* Put the values in the expression and return it.
+
+## Implementation
+
+Following are the implementation of the above mentioned problem:
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@Shivashish-rwt"/>
+
+```cpp
+class Solution {
+public:
+    void insertionSort(vector<int> &arr, int size_of_array) {
+        for (int i = 1; i < size_of_array; i++) {
+            int nums = arr[i]; // Picking element from array
+            int j = i - 1;
+
+            while (j >= 0 and arr[j] > nums) { // Comparing nums with all the elements left to it
+                arr[j + 1] = arr[j]; // Shifting the greater element to the right
+                j--;
+            }
+            arr[j + 1] = nums; // Placing selected element at correct position
+        }
+    }
+
+    int maxProduct(vector<int>& nums) {
+        insertionSort(nums,nums.size());
+        int n = nums.size();
+        return (nums[n - 1] - 1)*(nums[n - 2] - 1);
+    }
+};
+```
+
+</TabItem>
+
+<TabItem value="java" label="Java">
+<SolutionAuthor name="@Shivashish-rwt"/>
+
+```java
+class Solution {
+    void insertion_sort(int arr[],int size_of_array) {
+		for (int i = 1; i < size_of_array; ++i) {
+			int nums = arr[i]; // Picking element from array
+			int j = i - 1;
+
+			while (j >= 0 && arr[j] > nums) { // Comparing nums with all the elements left to it
+				arr[j + 1] = arr[j]; // Shifting the greater element to the right
+				j = j - 1;
+			}
+			arr[j + 1] = nums; // Placing selected element at correct position
+		}
+	}
+    public int maxProduct(int[] nums) {
+        insertion_sort(nums,nums.length);
+        int n = nums.length;
+        return (nums[n - 1] - 1)*(nums[n - 2] - 1);      
+    }
+}
+```
+
+</TabItem>
+
+<TabItem value="py" label="Python">
+<SolutionAuthor name="@Shivashish-rwt"/>
+
+```py
+class Solution:
+    def insertionSort(arr,size_of_array):
+        
+        for i in range(1, size_of_array):
+            nums = arr[i]  # Picking element from array
+            j = i - 1
+              
+            while j >= 0 and arr[j] > nums: # Comparing nums with all the elements left to it
+                arr[j + 1] = arr[j] # Shifting the greater element to the right
+                j = j - 1
+        
+            arr[j + 1] = nums # Placing selected element at correct position
+        
+
+    def maxProduct(self, nums: List[int]) -> int:
+        Solution.insertionSort(nums,len(nums))
+        return (nums[-1] - 1) * (nums[-2] - 1)
+```
+
+</TabItem>
+
+</Tabs>
+
+## Time Complexity
+
+Apart from sorting the the given array we are not doing anything in the solution. The expression would take $\mathcal{O}(1)$ to solve and insertion sort would take $\mathcal{O}(n^2)$ to sort the array.
```

Comment:
O(1) is not relavant here, the idea is to focus on Insertion sort and it's complexity. 

Until unless if any specific line cause, O(n) or O(log n) then you can mention it.